org.eclipse.jdt.core/model/org/eclipse/jdt/internal/core/JavaModel.java

```diff
@@ -337,6 +337,13 @@ public static Object getTarget(IPath path, boolean checkResourceExistence) {
 		return target;
 	return getExternalTarget(path, checkResourceExistence);
 }
+public static Object getTarget(IClasspathEntry entry, boolean checkResourceExistence) {
+	return getExternalTarget(entry.getPath(), checkResourceExistence);
```

Comment:
I believe this is not correct to skip `getWorkspaceTarget()` here and assume the classpath entry can't represent a path in the workspace.

Same below.